File: app/src/main/java/org/progresspics/android/EditableTextOverlay.java
package org.progresspics.android;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class EditableTextOverlay extends android.support.v7.widget.AppCompatEditText {

    public EditableTextOverlay(Context context) {
        super(context);
    }

    public EditableTextOverlay(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public EditableTextOverlay(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent event) {
        return isEnabled() && super.dispatchTouchEvent(event);
    }

    public void activate(boolean on) {
        setEnabled(on);
        setFocusable(on);
        setFocusableInTouchMode(on);
        setClickable(on);
    }
}

File: app/src/main/java/org/progresspics/android/EditableTextOverlay.kt
package org.progresspics.android

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent

class EditableTextOverlay : android.support.v7.widget.AppCompatEditText {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
        return isEnabled && super.dispatchTouchEvent(event)
    }

    fun activate(on: Boolean) {
        isEnabled = on
        isFocusable = on
        isFocusableInTouchMode = on
        isClickable = on
    }
}
